paasta_tools/smartstack_tools.py

```diff
@@ -168,7 +168,7 @@ def get_smartstack_replication_for_attribute(attribute, service, namespace, blac
     replication_info = {}
     filtered_slaves = mesos_tools.get_all_slaves_for_blacklist_whitelist(
         blacklist=blacklist,
-        whitelist=[],
+        whitelist=None,
```

Comment:
isn't it the code on the other side thats wrong? i.e. why aren't we passing [] here and the ``get_all_slaves_for_blacklist_whitelist`` function dealing with it correctly?

Reply:
`DeployWhitelist` is an `Optional`, so `None` is fine: https://github.com/Yelp/paasta/blob/master/paasta_tools/utils.py#L151

Reply:
actually, I think the type in utils.py is wrong

Reply:
...or the type is right, but the docstring on `slave_passes_whitelist` is wrong